🐞 `dagger do --help` panics

## Problem

When running `dagger do --help` on a valid plan, I see the following error:

```
$ dagger do --help
panic: event: Action cannot be empty

goroutine 1 [running]:
go.dagger.io/dagger/telemetry.(*Telemetry).Push(0xc00046ae70, {0x3?, 0xc003e3f9b8?}, {0x1493fd8?, 0xc002940b80?})
        /src/telemetry/telemetry.go:88 +0x11e
go.dagger.io/dagger/cmd/dagger/cmd.glob..func2(0x1cabcc0, {0xc000122200, 0x1, 0x1})
        /src/cmd/dagger/cmd/do.go:91 +0x70c
github.com/spf13/cobra.(*Command).execute(0x1cabcc0, {0xc000122200, 0x1, 0x1})
        /root/.cache/go-mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x1cac6c0)
        /root/.cache/go-mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /root/.cache/go-mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /root/.cache/go-mod/github.com/spf13/cobra@v1.4.0/command.go:895
go.dagger.io/dagger/cmd/dagger/cmd.Execute()
        /src/cmd/dagger/cmd/root.go:96 +0x478
main.main()
        /src/cmd/dagger/main.go:8 +0x17
```

When executing an action in the same plan, for example `dagger do build`, the action is executed normally with no issue.

## Dagger version

Devel, git commit `eaff2c63d3fd41d85bb6fc6eed4cb515613b7a18`.

--- a/cmd/dagger/cmd/do.go
+++ b/cmd/dagger/cmd/do.go
@@ -88,12 +88,6 @@ var doCmd = &cobra.Command{
 			os.Exit(1)
 		}
 
-		tm.Push(ctx, event.RunStarted{
-			Action: cue.MakePath(targetPath.Selectors()[1:]...).String(),
-			Args:   os.Args[1:],
-			Plan:   fmt.Sprintf("%#v", daggerPlan.Source().Cue()),
-		})
-
 		action := daggerPlan.Action().FindByPath(targetPath)
 
 		if action == nil {
@@ -142,6 +136,13 @@ var doCmd = &cobra.Command{
 			doHelpCmd(cmd, daggerPlan, action, actionFlags, targetPath, []string{})
 			os.Exit(0)
 		}
+
+		// Fire "run started" event once we know there is an action to run (ie. not calling --help)
+		tm.Push(ctx, event.RunStarted{
+			Action: cue.MakePath(targetPath.Selectors()[1:]...).String(),
+			Args:   os.Args[1:],
+			Plan:   fmt.Sprintf("%#v", daggerPlan.Source().Cue()),
+		})
 
 		if f := viper.GetString("log-format"); f == "tty" || f == "auto" && term.IsTerminal(int(os.Stdout.Fd())) {
 			tty, err = logger.NewTTYOutput(os.Stderr)

